client/src/components/DatasetMoveSamplesModal.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react'
+import { Box, Grid, RadioButton, Text } from 'grommet'
+import { config } from 'config'
+import { useResponsive } from 'hooks/useResponsive'
+import { Button } from 'components/Button'
+import { Modal, ModalBody } from 'components/Modal'
+import { WarningText } from 'components/WarningText'
+import { InfoText } from 'components/InfoText'
+import { Link } from 'components/Link'
+
+export const DatasetMoveSamplesModal = ({
+  label = 'Move Samples',
+  title = 'Move Samples to My Dataset',
+  modalDisabled = false,
+  appendDisabled = false
+}) => {
+  const [showing, setShowing] = useState(false)
+  const [checkedAppendSamples, setCheckedAppendSamples] = useState(false)
+  const [checkedReplaceSamples, setCheckedReplaceSamples] = useState(false)
+
+  const { responsive } = useResponsive()
+  const totalSamples = 34
+
+  const handleClick = () => {
+    setShowing(true)
+  }
+
+  const handleChangedCheckAppend = () => {
+    setCheckedAppendSamples(true)
+    setCheckedReplaceSamples(false)
+  }
+  const handleChangedCheckReplace = () => {
+    setCheckedReplaceSamples(true)
+    setCheckedAppendSamples(false)
+  }
+
+  return (
+    <>
+      <Button
+        aria-label={label}
+        flex="grow"
+        primary
+        label={label}
+        disabled={modalDisabled}
+        onClick={handleClick}
+      />
+      <Modal title={title} showing={showing} setShowing={setShowing}>
+        <ModalBody>
+          <Grid
+            columns={['auto']}
+            margin={{ bottom: '0' }}
+            pad={{ bottom: '0' }}
+          >
+            <Box>
+              <WarningText
+                lineBreak={false}
+                text={`There are ${totalSamples} samples in My Dataset`}
+                iconMargin="0"
+              />
```

Comment:
Regarding the spacing issue, to resolve this, we can modify the `WarningText` to accept new props while keeping the default values to support the current UIs.

Also, the font size should be `21px` (currently `16px`, a fixed value set by `WarningText`), and the icon size should be `24px` which can be adjusted using the `iconSize` props. 

For example: 
```suggestion
             <WarningText
                lineBreak={false}
                text={`There are ${totalSamples} samples in My Dataset`}
                iconMargin="0"
                iconSize="24px" // set icon size
                textMargin="4px" // a new prop for text spacing
                textSize="21px" // a new prop for text size
              />
```

And we can update the `WarningText` as follows:

```jsx
eexport const WarningText = ({
  lineBreak = true,
  link = '',
  linkLabel = '',
  iconColor = 'status-warning',
  iconMargin = { right: 'medium' },
  iconName = 'Warning',
  iconPad = { right: 'small' },
  iconSize = '16px',
  text = '',
  textMargin = '8px', // a new prop
  textSize = '16px',  // a new prop
  newTab = false,
  children
}) => {
  return (
    <Box
      align="center"
      direction="row"
      margin={{ top: 'small', bottom: 'medium' }}
    >
      <Box margin={iconMargin} pad={iconPad}>
        <Icon color={iconColor} size={iconSize} name={iconName} />
      </Box>
      <Paragraph size={textSize} margin={textMargin}> // pass them here
```

Reply:
Instead of trying to add a ton of props for every bit and bob in this component.  I would instead just leverage the use of children and not pass text, link, or newTab. Honestly, newTab should always be true and removed anyways but dont do it in this PR.

You can literally put whatever you want and style it correctly which is preferred to having many specific props for styling nested components. The same as how `InfoText` is used.

Reply:
This is a great point! As David suggested, instead of editing the `WarningText` component, let's leverage `children` to address the off-spacing issue with the text 👍 

For instance (we use either Grommet `Paragraph` or `Text` component to render text): 

```suggestion
             <WarningText
                lineBreak={false}
                iconMargin="0"
                iconSize="24px" // set the icon size here
              >
              <Text size="24px" margin="ADJUST_AS_NEEDED"> 
                   There are {totalSamples} samples in My Dataset
              </ Text>
```